addons/point_of_sale/tests/test_pos_basic_config.py

```diff
@@ -503,3 +503,256 @@ def test_correct_partner_on_invoice_receivables(self):
         self.assertEqual(len(other_customer_invoice_receivable_counterpart), 1, msg='there should one aggregated invoice receivable counterpart for self.other_customer')
         self.assertEqual(bool(other_customer_invoice_receivable_counterpart.full_reconcile_id), True, msg='the aggregated receivable for self.other_customer should be reconciled')
         self.assertEqual(other_customer_invoice_receivable_counterpart.balance, -200, msg='aggregated balance should be -200')
+
+    def test_refund_customer_reconcile(self):
+        """ Test return invoiced order
+
+                2 orders
+                - 2nd order is returned
+
+                Orders
+                ======
+                +------------------+----------+-----------+----------+-----+-------+
+                | order            | payments | invoiced? | product  | qty | total |
+                +------------------+----------+-----------+----------+-----+-------+
+                | order 1          | bank     | yes       | product1 |   3 |    30 |
+                |                  |          |           | product2 |   2 |    40 |
+                |                  |          |           | product3 |   1 |    30 |
+                +------------------+----------+-----------+----------+-----+-------+
+                | order 2          | bank     | yes       | product1 |   3 |    30 |
+                |                  |          |           | product2 |   2 |    40 |
+                |                  |          |           | product3 |   1 |    30 |
+                +------------------+----------+-----------+----------+-----+-------+
+                | order 3 (return) | bank     | yes       | product1 |  -3 |   -30 |
+                |                  |          |           | product2 |  -2 |   -40 |
+                |                  |          |           | product3 |  -1 |   -30 |
+                +------------------+----------+-----------+----------+-----+-------+
+
+                Expected Result
+                ===============
+                +---------------------+---------+
+                | account             | balance |
+                +---------------------+---------+
+                | sale (sales)        |    -200 |
+                | sale (refund)       |     100 |
+                | pos receivable bank |     100 |
+                +---------------------+---------+
+                | Total balance       |     0.0 |
+                +---------------------+---------+
```

Comment:
this is not correct, it should be
> pos receivable bank -> 100
> receivable -> -100

Reply:
I don't understand why not (-100 order 1, -100 order 2), 100 from refund and pos receivable bank is 100 (100 + 100 - 100).

Probably I am loosing something, but balance should be 0, isn't it?

Reply:
I think you were expecting this:

```
                +---------------------+---------+
                | account             | balance |
                +---------------------+---------+
                | sale (sales+refunds)|    -100 |
                | pos receivable bank |     100 |
                +---------------------+---------+
                | Total balance       |     0.0 |
                +---------------------+---------+
```

Reply:
I believe you are using the other tests as starting point in writing these new ones. The journal entry being described in those existing tests is the journal entry created when the pos.session is closed.

Now, in these new test cases, since all the orders are invoiced, we expect no sales line in the journal entry for the session. It should mainly contain receivable lines that will be reconciled with the receivables of the invoices. If you are wondering where the sales lines are, they are in the invoices' journal entries.

Reply:
Thanks for the explanation, now it has more sense for me :wink: